pkg/tagger/tagstore.go

```diff
@@ -75,6 +75,12 @@ func (s *tagStore) processTagInfo(info *collectors.TagInfo) error {
 
 	storedTags.Lock()
 	defer storedTags.Unlock()
+	_, found := storedTags.lowCardTags[info.Source]
```

Comment:
This makes the race condition way less unlikely, but in a multi-threaded environment it could still happen that `found == false` here while another thread is updating the cache. Using a lock would make this thread safe.

Reply:
it protect to write in //, but not fix the ordering of the execution of the goroutine that try to update this record